src/pages/resultsView/mutation/ResultsViewMutationMapperStore.ts

```diff
@@ -102,4 +103,19 @@ export default class ResultsViewMutationMapperStore extends MutationMapperStore
             this.getDiscreteCNACache
         );
     }
+
+    @computed
+    get isCanonicalTranscript(): boolean | undefined {
```

Comment:
should only return boolean.  checks for loading status should be handled outside of this check.

Reply:
if transcript dropdown is disabled(https://github.com/cBioPortal/cbioportal-frontend/pull/3211/files#diff-3a3df9949f97de541d314ef00760e1d2R62), then this `isCanonicalTranscript` should be undefined. So I return boolean | undefined here.

Reply:
Isn't it always the canonical transcript if the dropdown is disabled? It feels like this function should return true if the dropdown is disabled. Also the store should ideally only know about data not the state (enabled/disabled) of a UI component. So probably it is a good idea to move the `undefined` logic out of this function.